odoo/addons/test_http/tests/test_device.py

```diff
@@ -344,6 +345,73 @@ def test_deletion_specific_device(self):
         res = self.hit('2024-01-01 08:00:30', '/test_http/greeting-user?readonly=0', headers={'User-Agent': USER_AGENT_linux_firefox})
         self.assertIn('/web/login', res.url)
 
+    # --------------------
+    # FILESYSTEM REFLEXION
+    # --------------------
+
+    def _duplicate_session_without_creating_file(self, session, count):
```

Comment:
the session store is mocked inside `test_http` to use `MemorySessionStore`, so technically no file is created, what are you trying to achieve here?

Reply:
The goal was to create different devices (`res.device.logs`) for the user.

The method used to insert these records is `_update_device` which takes the request as a parameter (hence the use of `MockRequest`).

I wanted to use this method to realise the complete flow.

I agree that the use of `MockRequest` is not particularly appropriate here.
I have modified the function to insert the records directly from the model.